cfg/config.go

```diff
@@ -2,45 +2,200 @@ package cfg
 type Config struct {
-	Version       string `toml:"version"`
-	Project       string `toml:"project-name"`
-	BuildType     string `toml:"build-type"`
-	BuildFilePath string `toml:"build-file-path"`
+	Version       string
+	Project       string
+	BuildType     string
+	BuildFilePath string
+	RemoteURL     string
 
-	Remotes map[string]*RemoteVPS `toml:"remotes"`
+	remotes map[string]*RemoteVPS
 }
 
 // NewConfig sets up Inertia configuration with given properties
-func NewConfig(version, project, buildType, buildFilePath string) *Config {
+func NewConfig(version, project, buildType, buildFilePath, remoteURL string) *Config {
 	cfg := &Config{
 		Version:   version,
 		Project:   project,
 		BuildType: buildType,
-		Remotes:   make(map[string]*RemoteVPS),
+		RemoteURL: remoteURL,
+		remotes:   make(map[string]*RemoteVPS),
 	}
 	if buildFilePath != "" {
 		cfg.BuildFilePath = buildFilePath
 	}
 	return cfg
 }
 
-// Write writes configuration to Inertia config file at path. Optionally
+// NewConfigFromFiles loads configuration from given filepaths
+func NewConfigFromFiles(projectConfigPath string, remoteConfigPath string) (*Config, error) {
+	// Attempt to read files
+	projectBytes, err := ioutil.ReadFile(projectConfigPath)
+	if err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
+	remotesBytes, err := ioutil.ReadFile(remoteConfigPath)
+	if err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
+
+	// At least one file must exist
+	if remotesBytes == nil && projectBytes == nil {
+		return nil, errors.New(NoInertiaConfig)
+	}
+
+	var (
+		project = &common.InertiaProject{}
+		remotes = &InertiaRemotes{}
+	)
+
+	// If only project is present, construct config using project
+	if projectBytes != nil && remotesBytes == nil {
+		err = toml.Unmarshal(projectBytes, project)
+		if err != nil {
+			return nil, fmt.Errorf("project config error: %s", err.Error())
+		}
+		return NewConfigFromTOML(*project, InertiaRemotes{Version: project.Version})
+	}
+
+	// If only remotes is present, construct config using remotes
+	if projectBytes == nil && remotesBytes != nil {
+		err = toml.Unmarshal(remotesBytes, remotes)
+		if err != nil {
+			return nil, fmt.Errorf("remotes config error: %s", err.Error())
+		}
+		return NewConfigFromTOML(common.InertiaProject{
+			Version: remotes.Version,
+		}, *remotes)
+	}
+
+	// If both files are present, construct config using both
+	err = toml.Unmarshal(projectBytes, project)
+	if err != nil {
+		return nil, fmt.Errorf("project config error: %s", err.Error())
+	}
+	err = toml.Unmarshal(remotesBytes, remotes)
+	if err != nil {
+		return nil, fmt.Errorf("remotes config error: %s", err.Error())
+	}
+	return NewConfigFromTOML(*project, *remotes)
+}
+
+// NewConfigFromTOML loads configuration from TOML format structs
+func NewConfigFromTOML(project common.InertiaProject, remotes InertiaRemotes) (*Config, error) {
+	// Set remote defaults
+	if remotes.Remotes == nil {
+		r := make(map[string]*RemoteVPS)
+		remotes.Remotes = r
```

Comment:
Is there a reason for doing these separately? It doesn't seem like `r` gets used elsewhere